src/System.Text.Json/tests/Serialization/Value.ReadTests.NonGenericCollections.cs

```diff
@@ -304,8 +304,13 @@ public static void ReadPrimitiveStack()
             }
             Assert.Equal(0, count);
 
-            // TODO: use reflection to support types deriving from Stack.
-            Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<StackWrapper>(@"[1,2]"));
+            StackWrapper wrapper =  JsonSerializer.Deserialize<StackWrapper>(@"[1,2]");
```

Comment:
Does serialization + deserialization of Stacks and Queues round-trip like one would expect? (same for Json.NET)?

Please add a test or point me to one. Thanks

Reply:
Per offline discussion, we don't round-trip and neither does Json.NET meaning serialize()->deserialize()->serialize() will not result in the same state. I think this is incorrect.

@JamesNK can you chime in please. Thanks

Reply:
Json.NET reverses one or both of them (I can't remember off the top of my head which) when they're round-tripped.

If you pass the offending collection an enumerable in the order is was written and read then it will deserialized reversed.

People have complained about it, but I never fixed it for backwards compat.

Reply:
Queue round-trips for both System.Text.Json & Json.NET

We'd roundtrip if the serializer wrote stack JSON in the reverse order of the enumerator returned by stacks. Filing https://github.com/dotnet/corefx/issues/41887 to track this.